google/auth/aio/transport/aiohttp.py

```diff
@@ -14,3 +14,103 @@
 
 """Transport adapter for Asynchronous HTTP Requests.
 """
+
+import asyncio
+
+try:
+    import aiohttp
+except ImportError as caught_exc:  # pragma: NO COVER
+    raise ImportError(
+        "The aiohttp library is not installed from please install the aiohttp package to use the aiohttp transport."
+    ) from caught_exc
+from google.auth import exceptions
+
+_DEFAULT_TIMEOUT = 180  # in seconds
+
+
+class Request:
+    """Asynchronous Requests request adapter.
+
+    This class is used internally for making requests using asyncio transports
+    in a consistent way. If you use :class:`AuthorizedSession` you do not need
+    to construct or use this class directly.
+
+    This class can be useful if you want to manually refresh a
+    :class:`~google.auth.aio.credentials.Credentials` instance::
+
+        import google.auth.aio.transport.aiohttp
+
+        request = google.auth.aio.transport.aiohttp.Request()
+
+        await credentials.refresh(request)
+
+    Args:
+        session (aiohttp.ClientSession): An instance :class:`aiohttp.ClientSession` used
+            to make HTTP requests. If not specified, a session will be created.
+
+    .. automethod:: __call__
+    """
+
+    def __init__(self, session=None):
+        # TODO(ohmayr): Evaluate if we want auto_decompress=False.
+        # and if we want to update it in the passed in session.
+        self.session = session or aiohttp.ClientSession(auto_decompress=False)
```

Comment:
Let's just use the defaults, and if anyone needs to set a custom parameter they can pass in their own session.

What do you think?

Reply:
Sounds good to me. I think we can work around by not setting `auto_decompress` so it should be fine or come back to it later if always needed.

What I'm unsure about is why `trust_env` is set [here](https://github.com/googleapis/google-auth-library-python/blob/18c2ec1b571d506c0dbcffc483aa5e7b95e1b246/google/auth/transport/_aiohttp_requests.py#L311) as False.

By default the value is set to True i.e. it'll automatically pull up any proxy settings in the environment unlike what we've been doing in our experimental code.

Reply:
I'd like to correct myself here. The default value for `trust_env` is False.

will punt adding this as a parameter until required. Thanks!